File: Examples/VisualBasic/ProgrammersGuide/Working-Shapes/Working-with-Shapes-Gluing/GlueVisioShapes.vb
Imports VisualBasic
Imports Aspose.Diagram
Imports System

Public Class GlueVisioShapes
    Public Shared Sub Run()
        ' ExStart:GlueVisioShapes
        ' The path to the documents directory.
        Dim dataDir As String = RunExamples.GetDataDir_Shapes()

        ' Load diagram
        Dim diagram As New Diagram(dataDir & Convert.ToString("Drawing1.vsdx"))

        ' Get a particular page
        Dim page As Page = diagram.Pages.GetPage("Page-1")
        ' set shape id
        Dim shape1_ID As Long = 7
        Dim shape2_ID As Long = 494
        ' Glue shapes
        page.GlueShapes(shape1_ID, Aspose.Diagram.Manipulation.ConnectionPointPlace.Center, shape2_ID)

        ' Save diagram
        diagram.Save(dataDir & Convert.ToString("GlueVisioShapes_Out.vsdx"), SaveFileFormat.VSDX)
        ' ExEnd:GlueVisioShapes
    End Sub
End Class

File: Examples/CSharp/Working-Shapes/Working-with-Shapes-Gluing/GlueVisioShapes.cs
using Aspose.Diagram;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Aspose.Diagram.Examples.CSharp.Working_Shapes.Working_with_Shapes_Gluing
{
    public class GlueVisioShapes
    {
        public static void Run()
        {
            // ExStart:GlueVisioShapes
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_Shapes();

            // Load diagram
            Diagram diagram = new Diagram(dataDir + "Drawing1.vsdx");

            // Get a particular page
            Page page = diagram.Pages.GetPage("Page-1");
            // Set shape id
            long shape1_ID = 7;
            long shape2_ID = 494;
            // Glue shapes
            page.GlueShapes(shape1_ID, Aspose.Diagram.Manipulation.ConnectionPointPlace.Center, shape2_ID);

            // Save diagram
            diagram.Save(dataDir + "GlueVisioShapes_Out.vsdx", SaveFileFormat.VSDX);
            // ExEnd:GlueVisioShapes
        }
    }
}
